### PR TITLE
Add GetBucketReplicationMetrics api

### PR DESCRIPTION
**Pull request by poornas**

### CODE REVIEW

File: pkg/replication/replication.go [hunk 1]
```diff
@@ -604,7 +604,22 @@ func (s SourceSelectionCriteria) Validate() error {
...
+	// Pending size in bytes
+	PendingSize uint64 `json:"objectsPendingReplicationTotalSize"`
+	// Completed size in bytes
+	ReplicatedSize uint64 `json:"objectsReplicatedTotalSize"`
+	// Total Replica size in bytes
+	ReplicaSize uint64 `json:"objectsReplicaTotalSize"`
+	// Failed size in bytes
+	FailedSize uint64 `json:"objectsFailedReplicationTotalSize"`
+	// Total number of pending operations including metadata updates
+	OperationsPendingCount uint64 `json:"objectsPendingReplicationCount"`
```
Comment (harshavardhana) [inline pkg/replication/replication.go:624]:
```suggestion
	PendingCount uint64 `json:"pendingReplicationCount"`
```

File: pkg/replication/replication.go [hunk 2]
```diff
@@ -604,7 +604,22 @@ func (s SourceSelectionCriteria) Validate() error {
...
+		return fmt.Errorf("invalid ReplicaModification status")
 	}
 	return nil
 }
+
+// Metrics represents inline replication metrics
+// such as pending, failed and completed bytes in total for a bucket
+type Metrics struct {
+	// Pending size in bytes
+	PendingSize uint64 `json:"objectsPendingReplicationTotalSize"`
```
Comment (harshavardhana) [inline pkg/replication/replication.go:616]:
```suggestion
	PendingSize uint64 `json:"pendingReplicationSize"`
```

File: pkg/replication/replication.go [hunk 3]
```diff
@@ -604,7 +604,22 @@ func (s SourceSelectionCriteria) Validate() error {
...
+	// Completed size in bytes
+	ReplicatedSize uint64 `json:"objectsReplicatedTotalSize"`
```
Comment (harshavardhana) [inline pkg/replication/replication.go:618]:
```suggestion
	ReplicatedSize uint64 `json:"completedReplicatedSize"`
```

File: pkg/replication/replication.go [hunk 4]
```diff
@@ -604,7 +604,22 @@ func (s SourceSelectionCriteria) Validate() error {
...
+	// Total Replica size in bytes
+	ReplicaSize uint64 `json:"objectsReplicaTotalSize"`
```
Comment (harshavardhana) [inline pkg/replication/replication.go:620]:
```suggestion
	ReplicaSize uint64 `json:"replicaSize"`
```

File: pkg/replication/replication.go [hunk 5]
```diff
@@ -604,7 +604,22 @@ func (s SourceSelectionCriteria) Validate() error {
...
+	// Failed size in bytes
+	FailedSize uint64 `json:"objectsFailedReplicationTotalSize"`
```
Comment (harshavardhana) [inline pkg/replication/replication.go:622]:
```suggestion
	FailedSize uint64 `json:"failedReplicationSize"`
```

File: pkg/replication/replication.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/replication/replication.go:624]: We also need FailedCount to calculate failure rate

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**